### ISSUE TITLE
Subtyping problem

### ISSUE THREAD

**Issue by smimram** [savonet/liquidsoap#2092]
The following bug is ennoying:
```
s = playlist("~/Music")
t = sine()
s = add([s, t])
```
raises
```
At ./bug.liq, line 8, char 12:
s = add([s, t])

Error 5: this value has no method length
```

However, the following simpler toy example goes though:
```
r = 4.{a = "A", b = "B", c = "C"}
s = 2.{a = "O", x = "X"}
_ = [r, s]
_ = [s, r]
```

**Comment by smimram** [comment]
Shorter version:
```
def playlist()
  s = playlist.list([])
  # s = sine()
  def s.length() =
    0
  end
  s
end

s = playlist()
t = sine()
_ = [s, t]
```
Note that if I define `s` to `sine()` we don't get the bug anymore.

**Comment by smimram** [comment]
For info, it seems to come from the fact that both sources define methods `remaining` but with different types.

**Comment by smimram** [comment]
Fixed in 1d87e7660d4f918bd2285b3bb75f8e76cdbcccd0.
